diff --git a/profiles/openberkeley/modules/ucb/ucberkeley_cas/ucberkeley_cas.features.defaultconfig.inc b/profiles/openberkeley/modules/ucb/ucberkeley_cas/ucberkeley_cas.features.defaultconfig.inc
index 8d7b697..6e53e1b 100644
--- a/profiles/openberkeley/modules/ucb/ucberkeley_cas/ucberkeley_cas.features.defaultconfig.inc
+++ b/profiles/openberkeley/modules/ucb/ucberkeley_cas/ucberkeley_cas.features.defaultconfig.inc
@@ -86,12 +86,9 @@ function ucberkeley_cas_defaultconfig_strongarm() {
   $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
   $strongarm->api_version = 1;
   $strongarm->name = 'cas_auto_assigned_role';
+  $rid_auth_user = array_search('authenticated user', user_roles());
   $strongarm->value = array(
-    2 => TRUE,
-    4 => 0,
-    5 => 0,
-    6 => 0,
-    7 => 0,
+    $rid_auth_user => TRUE,
   );
   $export['cas_auto_assigned_role'] = $strongarm;
 
@@ -99,7 +96,7 @@ function ucberkeley_cas_defaultconfig_strongarm() {
   $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
   $strongarm->api_version = 1;
   $strongarm->name = 'cas_cert';
-  $strongarm->value = '/usr/local/share/certs/cacert.pem';
+  $strongarm->value = '';
   $export['cas_cert'] = $strongarm;
 
   $strongarm = new stdClass();
